src/oatpp/core/async/Processor.hpp

```diff
@@ -108,8 +112,19 @@ class Processor {
 
 private:
 
-  bool m_running = true;
-  std::atomic<v_int32> m_tasksCounter;
+  std::atomic_bool m_running{true};
+  std::atomic<v_int32> m_tasksCounter{0};
+
+private:
+
+  std::recursive_mutex m_coroutineWaitListsWithTimeoutsMutex;
```

Comment:
Do we need this mutex to be recursive?
At a first glance, it seems we can have it as a simple `std::mutex`.

Reply:
I made it a recursive_mutex, because in `Processor::checkCoroutinesForTimeouts()` we call `CoroutineWaitList::checkCoroutinesForTimeouts()` which in return could remove the current WaitList from the list of coroutinewaitlists with timeout from the processor. Becaus we need to hold the lock of `m_coroutineWaitListsWithTimeoutsMutex` while we are inside `Processor::checkCoroutinesForTimeouts()`  it has to be recursive. 

In the destructor of `CoroutineWaitList` we would remove the waitlist from the processors list as well. I played with the idea of having a second list with "to-remove-waitlists" and at the begining of `Processor::checkCoroutinesForTimeouts()` work through this list and merge it with the actual `m_coroutineWaitListsWithTimeouts`. But this would introduce a race condition where the `CoroutineWaitList` is already destroy while the `Processors` still holds a pointer to the now deleted object and calls the `CoroutineWaitList::checkCoroutinesForTimeouts()` method.